File: Provas/2016.1/questao1-java/LogDecorator.java
public class LogDecorator extends Decorator
{
    public LogDecorator(IComponent decorated) {
        super(decorated);
    }
    public void aumentarPreco(double percentual) {
        super.aumentarPreco(percentual);
        System.out.println("Fazendo o log");
    }
}

File: Provas/2016.1/questao1-qt/logdecorator.cpp
#include "logdecorator.h"

#include <QDebug>

LogDecorator::LogDecorator(IComponent *decorated)
    : Decorator(decorated)
{
}
void LogDecorator::aumentarPreco(double percentual) {
    qDebug() << "Fazendo log";
    Decorator::aumentarPreco(percentual);
}
